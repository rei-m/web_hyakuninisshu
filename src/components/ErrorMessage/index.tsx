import * as React from 'react';
import styled from '@src/styles/styled-components';

export interface Props {
  text: string;
}

const Container = styled.div`
  height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Message = styled.p`
  font-size: 1.6rem;
`;

const ErrorMessage: React.FC<Props> = ({ text }) => (
  <Container>
    <Message>{text}</Message>
  </Container>
);

export default ErrorMessage;
