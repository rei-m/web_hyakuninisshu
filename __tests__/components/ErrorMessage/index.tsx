import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import ErrorMessage, { Props } from '@src/components/atoms/ErrorMessage';

describe('<ErrorMessage />', () => {
  let baseProps: Props;

  beforeEach(() => {
    baseProps = {
      text: 'Test',
    };
  });

  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<ErrorMessage {...baseProps} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
