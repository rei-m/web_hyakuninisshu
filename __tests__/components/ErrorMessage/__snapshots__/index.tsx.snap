// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render component 1`] = `
<div
  className="sc-bdVaJa gPmDrA"
>
  <p
    className="sc-bwzfXH jQIpiU"
  >
    Test
  </p>
</div>
`;
